fuel-merkle/src/binary/root_calculator.rs

```diff
@@ -32,6 +32,20 @@ impl MerkleRootCalculator {
         Self { stack }
     }
 
+    pub fn new_from_existing_leafs(leaf_hashes: Vec<Bytes32>) -> Self {
+        let mut calculator = Self::new();
+        for leaf in leaf_hashes {
+            calculator
+                .push_with_callback::<_, Infallible>(
+                    Node::create_leaf_with_hash(0, leaf)
+                        .expect("Zero is a valid index for a leaf"),
+                    |_| Ok(()),
+                )
+                .expect("Tree too large");
+        }
```

Comment:
```suggestion
     leaf_hashes.into_iter().for_each(|leaf| {
        calculator
            .push_with_callback::<_, Infallible>(
                Node::create_leaf_with_hash(0, leaf)
                    .expect("Zero is a valid index for a leaf"),
                |_| Ok(()),
            )
            .expect("Tree too large");
    });
```

Reply:
Why this syntax is better for you ?

Reply:
doesn't matter too much, non blocking suggestion haha 

if you want to make the function signature compatible with the `root` function used you can use an iterator

Reply:
https://github.com/FuelLabs/fuel-vm/pull/896#discussion_r1930706927